dvc/schema.py

```diff
@@ -34,7 +35,24 @@
         Optional(StageParams.PARAM_LOCKED): bool,
         Optional(StageParams.PARAM_META): object,
         Optional(StageParams.PARAM_ALWAYS_CHANGED): bool,
-        **{Optional(p.value): [str] for p in OutputParams},
+        Optional(StageParams.PARAM_OUTS): [
+            Any(
+                str,
+                {
+                    str: {
+                        Optional(BaseOutput.PARAM_CACHE): bool,
+                        Optional(BaseOutput.PARAM_PERSIST): bool,
+                    }
+                },
+            )
+        ],
+        Optional(StageParams.PARAM_OUTS_NO_CACHE): [str],
+        Optional(StageParams.PARAM_METRICS): [
+            Any(str, {str: {Optional(BaseOutput.PARAM_CACHE): bool}})
```

Comment:
Not supporting persist for metrics because it is the way we currently generate it during the serialization. Plus we don't really have a CLI option for it. But seems like `persist`ent metrics does make sense, especially when we talk about plots that might accumulate data from run to run. But let's wait for someone to ask for it.

Note that in old dvc-files that(persistent metrics) was supported if user set `metric: True, persist=True` by hand.